nmostesting/ControllerTest.py

```diff
@@ -117,12 +124,17 @@ def set_up_tests(self):
         # Reset registry to clear previous heartbeats, etc.
         self.primary_registry.reset()
         self.primary_registry.enable()
-        self.mock_registry_base_url = 'http://' + get_default_ip() + ':' + \
-            str(self.primary_registry.get_data().port) + '/'
-        self.mock_node_base_url = 'http://' + get_default_ip() + ':' + str(self.node.port) + '/'
+        if CONFIG.ENABLE_HTTPS:
+            self.mock_registry_base_url = 'https://' + CONFIG.MOCKS_HOSTNAME + ':' + \
+                str(self.primary_registry.get_data().port) + '/'
+            self.mock_node_base_url = 'https://' + CONFIG.MOCKS_HOSTNAME + ':' + str(self.node.port) + '/'
+        else:
+            self.mock_registry_base_url = 'http://' + get_default_ip() + ':' + \
+                str(self.primary_registry.get_data().port) + '/'
+            self.mock_node_base_url = 'http://' + get_default_ip() + ':' + str(self.node.port) + '/'
```

Comment:
Seems like there may be a more elegant/maintainable version of this code struggling to get out... :-)

E.g. use `self.protocol`

Reply:
Yes, agreed :) Is this any better (assuming that the mocks_hosthame is calculated earlier)?
```suggestion
        host = mocks_hostname if CONFIG.ENABLE_HTTPS else get_default_ip()
        self.mock_registry_base_url = self.protocol + '://' + host + ':' + str(self.primary_registry.get_data().port) + '/'
        self.mock_node_base_url = self.protocol + '://'  + host + ':' + str(self.node.port) + '/'
```

Reply:
Yep, that's the sort of thing 👍

Reply:
Code updated as per my comment above.